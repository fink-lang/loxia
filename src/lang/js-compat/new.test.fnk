{fink2js} = import '../../testing'
{describe, it, eq, snapshot} = import '../../testing/jest'


describe:: 'new', fn:
  it:: 'compiles', fn:
    fink2js(`
      foo = new Set()
      bar = new Set:: 'abca'
    `) eq snapshot


