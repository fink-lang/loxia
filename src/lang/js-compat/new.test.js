import {fink2js} from '../../testing';


describe('new', ()=> {
  it('transforms new', ()=> {
    expect(
      fink2js(`
        foo = new Set()
      `)
    ).toMatchSnapshot();
  });
});
