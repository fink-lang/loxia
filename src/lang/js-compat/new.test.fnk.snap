// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new compiles 1`] = `
"const foo = new Set();
const bar = new Set(\`abca\`);
Object.assign(module.exports, {
  foo,
  bar
});"
`;
